### PR TITLE
Fix New-TemporaryFile online help URI

### LINKED ISSUE #6186: Cmdlet Help link broken for New-TemporaryFile
The cmdlet help link for `New-TemporaryFile` is broken.

Steps to reproduce
------------------

```powershell
Get-Help New-TemporaryFile -Online
```

Expected behavior
-----------------

I get redirected to a help page like e.g. to
https://docs.microsoft.com/powershell/module/Microsoft.PowerShell.Utility/New-TemporaryFile?view=powershell-6.0

Actual behavior
---------------

I get redirected to
https://technet.microsoft.com/library/cc732148.aspx
image

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.0-preview.582
PSEdition                      Core
GitCommitId                    v6.1.0-preview.582
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by adityapatwardhan**
Fixes #6186 

## PR Summary

Fix the link for online help content for New-TemporaryFile.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace Microsoft.PowerShell.Commands
     /// <summary>
     /// The implementation of the "New-TemporaryFile" cmdlet
     /// </summary>
-    [Cmdlet(VerbsCommon.New, "TemporaryFile", SupportsShouldProcess = true, HelpUri = "https://go.microsoft.com/fwlink/?LinkId=526726")]
+    [Cmdlet(VerbsCommon.New, "TemporaryFile", SupportsShouldProcess = true, HelpUri = "https://go.microsoft.com/fwlink/?LinkId=821836")]
```
Comment (dantraMSFT) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs]: This like resolves to PowerShell 5.1 (https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Utility/New-TemporaryFile?view=powershell-5.1)  There is a PowerShell 6 page for this cmdlet (https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Utility/New-TemporaryFile?view=powershell-6)

**Review by dantraMSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs (hunk 1 above)
Comment (joeyaiello) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs]: So the new fwlink points to 5.1, and IMO it should probably point to the version-agnostic link (for which we can change the default later). This way, when we update to 6, we can rev the default for *all* modules on docs.microsoft.com to 6.1, and then we don't have to change the fwlinks.

However, if we go that route, we need to make sure that 6 has distinct fwlinks from 5.1, and i'm not sure we've got that work fully understood or costed (that's a *lot* of fwlinks).

**Comment by adityapatwardhan** [comment]
@dantraMSFT  The link now forwards to v6. Can you have another look?

**Review by dantraMSFT — APPROVED** [review]

**Comment by iSazonov** [comment]
Is the PR ready to merge?

**Comment by anmenaga** [comment]
@TravisEz13 Ready for Merge.

**Comment by adityapatwardhan** [comment]
@TravisEz13 ok to merge?

**Pull request merged**